pkg/cmd/project/shared/format/json.go

```diff
@@ -9,7 +9,7 @@ import (
 
 // JSONProject serializes a Project to JSON.
 func JSONProject(project queries.Project) ([]byte, error) {
```

Comment:
All these `JSON...` functions I think can and should be refactored to utilize the `ExportData` pattern we have established else were in the codebase. For now just doing it for `Project` and `Projects` but doing it for other exported structs through the `project` commands.  

For example 
```go

//TODO: implement in queries.go near definition of Project
func (p Project) ExportData(_ []string) map[string]interface{} {
  return nil
}

//TODO: implement in queries.go near definition of Projects
func (p Projects) ExportData(_ []string) map[string]interface{} {
  return nil
}

func JSONProject(project []queries.Project) ([]byte, error) {
    return json.Marshal(project.ExportData(nil))
}

func JSONProject(projects []queries.Projects) ([]byte, error) {
    return json.Marshal(projects.ExportData(nil))
}
```

Basically I think all these structs should know how to serialize themselves into maps rather than having these helper function that do it. We can keep the helper functions in place for now but eventually I see them going away completely.

This does seem like a large refactor job though, so willing to hear differing opinions on this.

Reply:
That'd simplify things quite a bit. I was a little surprised to see a near-duplication of models, and this would clean it up quite a bit.

Reply:
I started on this but, you're right: it's a big refactoring. Perhaps in another PR just to isolate all the changes? I stashed away a good start on this for now.